scratchwork/main.go

```diff
@@ -33,13 +34,48 @@ var (
 	// todo udpate the legendHost based on the legendsPort (when it changes)
 	// url.Parse -- but I personally find this code hard to read.
 	legendHost = fmt.Sprintf("http://localhost:%d", legendsPort)
+
+	cardCodeToName = make(map[string]string)
 )
 
+type PositionalRectangles struct {
+	GameState   string
+	Rectangles  Rectangles
+}
+
+type Rectangles []struct {
+	CardCode    string
+	LocalPlayer bool
+}
+
 type StaticDeckList struct {
 	DeckCode    string
 	CardsInDeck map[string]int
 }
 
+type Cards []struct {
+	CardCode    string
+	CardName    string `json:"name"`
+}
+
+func fillCardCodeToNameMap() error{
```

Comment:
this functionality, I think, can be improved inone of two ways:

1. make the map that it writes to non-global
2. make the map somehow read only and only fill it out at the start of the process.

The reason for the above is the following:
1. maps are not thread safe. therefore using go routines its easy to write bugs that will crash the process
2. there's a lot of work that we do here every time that we don't need to do, so at minimum it should be cached

Reply:
```
type readonlymap interface {
  Get(key string) value interface{}
}

func init() {
  // this gets called when the package is imported and only once.
  // make map here and only allow it to be read after this
}
```

Reply:
```
func fillCardCodeToNameMap() error{
  if len(cardCodeToName) > 0 {
    return nil
  }
  ...
}
```